PyInstaller/depend/analysis.py

```diff
@@ -764,12 +764,38 @@ def get_bootstrap_modules():
         mod = __import__(mod_name)  # C extension.
         if hasattr(mod, '__file__'):
             loader_mods.append((mod_name, os.path.abspath(mod.__file__), 'EXTENSION'))
+
+    # modify a copy of pyiboot01_bootstrap by related arguments
+    pyiboot01_name = 'pyiboot01_bootstrap.py'
+    pyiboot01_preserved_envs = pyiboot01_preserved_envs or []
+    pyiboot01_source = os.path.join(loaderpath, pyiboot01_name)
+    pyiboot01_context = dict(
+        pyiboot01_preserved_envs=pyiboot01_preserved_envs,
+    )
+    logger.info('Building %s with %r', pyiboot01_name, pyiboot01_context)
+    with open(pyiboot01_source, 'r') as f:
+        pyiboot01_content = f.read()
+    pyiboot01_content = render_pyfile_template(pyiboot01_content, pyiboot01_context)
+    pyiboot01_built = os.path.join(workpath, pyiboot01_name)
+    with open(pyiboot01_built, 'w') as f:
+        f.write(pyiboot01_content)
+
```

Comment:
This sould go into a function/method of its own, to make the code easier to understand.

Reply:
This could go into `render_pyfile_template(template_filename, output_directory)`:
```
pyiboot01_context = dict(
   preserved_envs=preserved_envs or []
)
render_pyfile_template(os.path.join(loaderpath, 'pyiboot01_bootstrap.py'),
                       workpath)
```
and render_pyfile_template will log the message, read the file, write the output file (basename of template).

What do you think?

Reply:
Update as suggested